### PR TITLE
crimson/net: misc fixes in v1 read path

### PR DESCRIPTION
**Pull request by cyx1231st**
https://github.com/ceph/ceph/pull/27809#discussion_r279225230

### CODE REVIEW

File: src/crimson/net/ProtocolV1.cc [hunk 1]
```diff
@@ -777,21 +777,23 @@ seastar::future<> ProtocolV1::read_message()
...
+          throw std::system_error{make_error_code(error::negotiation_failure)};
+        }
       }
       // TODO: set time stamps
       msg->set_byte_throttler(conn.policy.throttler_bytes);
 
-      if (!conn.update_rx_seq(msg->get_seq())) {
+      if (unlikely(!conn.update_rx_seq(msg->get_seq()))) {
         // skip this message
+        msg->put();
```
Comment (tchaikov) [inline src/crimson/net/ProtocolV1.cc]: can we move `auto msg_ref = MessageRef{msg, add_ref};` at line 800 (802 in this patch) after the `if` block of `if (unlikely(!msg)) {`?

so we don't need to call `msg->put()` explicitly in the error handling paths.

Comment (cyx1231st) [reply src/crimson/net/ProtocolV1.cc]: Makes sense, fixed.

File: src/crimson/net/ProtocolV1.cc [hunk 2]
```diff
@@ -777,27 +777,28 @@ seastar::future<> ProtocolV1::read_message()
       ::decode(m.footer, p);
       auto msg = ::decode_message(nullptr, 0, m.header, m.footer,
                                   m.front, m.middle, m.data, nullptr);
-      if (!msg) {
-	logger().debug("decode message failed");
-	return;
+      if (unlikely(!msg)) {
+        logger().warn("{} decode message failed", conn);
+        throw std::system_error{make_error_code(error::negotiation_failure)};
```
Comment (tchaikov) [inline src/crimson/net/ProtocolV1.cc]: can we define a new error for corrupted message?

Comment (cyx1231st) [reply src/crimson/net/ProtocolV1.cc]: done.

**Pull request merged**